# oncokbPrivateApiClient

OncoKB Private APIs
- API version: v1.2.1

These endpoints are for private use only.

  For more information, please visit [https://www.oncokb.org](https://www.oncokb.org)

*Automatically generated by the [Swagger Codegen](https://github.com/swagger-api/swagger-codegen)*


## Requirements

Building the API client library requires:
1. Java 1.7+
2. Maven/Gradle

## Installation

To install the API client library to your local Maven repository, simply execute:

```shell
mvn clean install
```

To deploy it to a remote Maven repository instead, configure the settings of the repository and execute:

```shell
mvn clean deploy
```

Refer to the [OSSRH Guide](http://central.sonatype.org/pages/ossrh-guide.html) for more information.

### Maven users

Add this dependency to your project's POM:

```xml
<dependency>
  <groupId>org.oncokb.client</groupId>
  <artifactId>oncokbPrivateApiClient</artifactId>
  <version>1.2.1-SNAPSHOT</version>
  <scope>compile</scope>
</dependency>
```

### Gradle users

Add this dependency to your project's build file:

```groovy
compile "org.oncokb.client:oncokbPrivateApiClient:1.2.1-SNAPSHOT"
```

### Others

At first generate the JAR by executing:

```shell
mvn clean package
```

Then manually install the following JARs:

* `target/oncokbPrivateApiClient-1.2.1-SNAPSHOT.jar`
* `target/lib/*.jar`

## Getting Started

Please follow the [installation](#installation) instruction and execute the following Java code:

```java

import org.oncokb.*;
import org.oncokb.auth.*;
import org.oncokb.client.*;
import org.oncokb.client.AnnotationsApi;

import java.io.File;
import java.util.*;

public class AnnotationsApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();
        
        // Configure API key authorization: Bearer
        ApiKeyAuth Bearer = (ApiKeyAuth) defaultClient.getAuthentication("Bearer");
        Bearer.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //Bearer.setApiKeyPrefix("Token");

        AnnotationsApi apiInstance = new AnnotationsApi();
        String copyNameAlterationType = "copyNameAlterationType_example"; // String | Copy number alteration type
        String hugoSymbol = "hugoSymbol_example"; // String | The gene symbol used in Human Genome Organisation. Example: BRAF
        Integer entrezGeneId = 56; // Integer | The entrez gene ID. (Higher priority than hugoSymbol). Example: 673
        String referenceGenome = "GRCh37"; // String | Reference genome, either GRCh37 or GRCh38. The default is GRCh37
        String tumorType = "tumorType_example"; // String | OncoTree(http://oncotree.info) tumor type name. The field supports OncoTree Code, OncoTree Name and OncoTree Main type. Example: Melanoma
        String evidenceType = "evidenceType_example"; // String | Evidence type to compute. This could help to improve the performance if you only look for sub-content. Example: ONCOGENIC. All available evidence type are GENE_SUMMARY, MUTATION_SUMMARY, TUMOR_TYPE_SUMMARY, PROGNOSTIC_SUMMARY, DIAGNOSTIC_SUMMARY, ONCOGENIC, MUTATION_EFFECT, PROGNOSTIC_IMPLICATION, DIAGNOSTIC_IMPLICATION, STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY, STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_RESISTANCE, INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY, INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_RESISTANCE. For multiple evidence types query, use ',' as separator.
        try {
            IndicatorQueryResp result = apiInstance.annotateCopyNumberAlterationsGetUsingGET(copyNameAlterationType, hugoSymbol, entrezGeneId, referenceGenome, tumorType, evidenceType);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AnnotationsApi#annotateCopyNumberAlterationsGetUsingGET");
            e.printStackTrace();
        }
    }
}

```

## Documentation for API Endpoints

All URIs are relative to *https://localhost:8080/oncokb-curate/api/v1*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*AnnotationsApi* | [**annotateCopyNumberAlterationsGetUsingGET**](docs/AnnotationsApi.md#annotateCopyNumberAlterationsGetUsingGET) | **GET** /annotate/copyNumberAlterations | annotateCopyNumberAlterationsGet
*AnnotationsApi* | [**annotateCopyNumberAlterationsPostUsingPOST**](docs/AnnotationsApi.md#annotateCopyNumberAlterationsPostUsingPOST) | **POST** /annotate/copyNumberAlterations | annotateCopyNumberAlterationsPost
*AnnotationsApi* | [**annotateMutationsByGenomicChangeGetUsingGET**](docs/AnnotationsApi.md#annotateMutationsByGenomicChangeGetUsingGET) | **GET** /annotate/mutations/byGenomicChange | annotateMutationsByGenomicChangeGet
*AnnotationsApi* | [**annotateMutationsByGenomicChangePostUsingPOST**](docs/AnnotationsApi.md#annotateMutationsByGenomicChangePostUsingPOST) | **POST** /annotate/mutations/byGenomicChange | annotateMutationsByGenomicChangePost
*AnnotationsApi* | [**annotateMutationsByHGVSgGetUsingGET**](docs/AnnotationsApi.md#annotateMutationsByHGVSgGetUsingGET) | **GET** /annotate/mutations/byHGVSg | annotateMutationsByHGVSgGet
*AnnotationsApi* | [**annotateMutationsByHGVSgPostUsingPOST**](docs/AnnotationsApi.md#annotateMutationsByHGVSgPostUsingPOST) | **POST** /annotate/mutations/byHGVSg | annotateMutationsByHGVSgPost
*AnnotationsApi* | [**annotateMutationsByProteinChangeGetUsingGET**](docs/AnnotationsApi.md#annotateMutationsByProteinChangeGetUsingGET) | **GET** /annotate/mutations/byProteinChange | annotateMutationsByProteinChangeGet
*AnnotationsApi* | [**annotateMutationsByProteinChangePostUsingPOST**](docs/AnnotationsApi.md#annotateMutationsByProteinChangePostUsingPOST) | **POST** /annotate/mutations/byProteinChange | annotateMutationsByProteinChangePost
*AnnotationsApi* | [**annotateStructuralVariantsGetUsingGET**](docs/AnnotationsApi.md#annotateStructuralVariantsGetUsingGET) | **GET** /annotate/structuralVariants | annotateStructuralVariantsGet
*AnnotationsApi* | [**annotateStructuralVariantsPostUsingPOST**](docs/AnnotationsApi.md#annotateStructuralVariantsPostUsingPOST) | **POST** /annotate/structuralVariants | annotateStructuralVariantsPost
*CancerGenesApi* | [**utilsAllActionableVariantsGetUsingGET**](docs/CancerGenesApi.md#utilsAllActionableVariantsGetUsingGET) | **GET** /utils/allActionableVariants | utilsAllActionableVariantsGet
*CancerGenesApi* | [**utilsAllActionableVariantsTxtGetUsingGET**](docs/CancerGenesApi.md#utilsAllActionableVariantsTxtGetUsingGET) | **GET** /utils/allActionableVariants.txt | utilsAllActionableVariantsTxtGet
*CancerGenesApi* | [**utilsAllAnnotatedVariantsGetUsingGET**](docs/CancerGenesApi.md#utilsAllAnnotatedVariantsGetUsingGET) | **GET** /utils/allAnnotatedVariants | utilsAllAnnotatedVariantsGet
*CancerGenesApi* | [**utilsAllAnnotatedVariantsTxtGetUsingGET**](docs/CancerGenesApi.md#utilsAllAnnotatedVariantsTxtGetUsingGET) | **GET** /utils/allAnnotatedVariants.txt | utilsAllAnnotatedVariantsTxtGet
*CancerGenesApi* | [**utilsAllCuratedGenesGetUsingGET**](docs/CancerGenesApi.md#utilsAllCuratedGenesGetUsingGET) | **GET** /utils/allCuratedGenes | utilsAllCuratedGenesGet
*CancerGenesApi* | [**utilsAllCuratedGenesTxtGetUsingGET**](docs/CancerGenesApi.md#utilsAllCuratedGenesTxtGetUsingGET) | **GET** /utils/allCuratedGenes.txt | utilsAllCuratedGenesTxtGet
*CancerGenesApi* | [**utilsCancerGeneListGetUsingGET**](docs/CancerGenesApi.md#utilsCancerGeneListGetUsingGET) | **GET** /utils/cancerGeneList | utilsCancerGeneListGet
*CancerGenesApi* | [**utilsCancerGeneListTxtGetUsingGET**](docs/CancerGenesApi.md#utilsCancerGeneListTxtGetUsingGET) | **GET** /utils/cancerGeneList.txt | utilsCancerGeneListTxtGet
*ClassificationApi* | [**classificationVariantsGetUsingGET**](docs/ClassificationApi.md#classificationVariantsGetUsingGET) | **GET** /classification/variants | classificationVariantsGet
*DrugsApi* | [**drugsGetUsingGET**](docs/DrugsApi.md#drugsGetUsingGET) | **GET** /drugs | Get all curated drugs.
*DrugsApi* | [**drugsLookupGetUsingGET**](docs/DrugsApi.md#drugsLookupGetUsingGET) | **GET** /drugs/lookup | Search drugs.
*EvidencesApi* | [**evidencesLookupGetUsingGET**](docs/EvidencesApi.md#evidencesLookupGetUsingGET) | **GET** /evidences/lookup | evidencesLookupGet
*EvidencesApi* | [**evidencesLookupPostUsingPOST**](docs/EvidencesApi.md#evidencesLookupPostUsingPOST) | **POST** /evidences/lookup | evidencesLookupPost
*EvidencesApi* | [**evidencesUUIDGetUsingGET**](docs/EvidencesApi.md#evidencesUUIDGetUsingGET) | **GET** /evidences/{uuid} | evidencesUUIDGet
*EvidencesApi* | [**evidencesUUIDsGetUsingPOST**](docs/EvidencesApi.md#evidencesUUIDsGetUsingPOST) | **POST** /evidences | evidencesUUIDsGet
*GenesApi* | [**genesEntrezGeneIdEvidencesGetUsingGET**](docs/GenesApi.md#genesEntrezGeneIdEvidencesGetUsingGET) | **GET** /genes/{entrezGeneId}/evidences | genesEntrezGeneIdEvidencesGet
*GenesApi* | [**genesEntrezGeneIdGetUsingGET**](docs/GenesApi.md#genesEntrezGeneIdGetUsingGET) | **GET** /genes/{entrezGeneId} | genesEntrezGeneIdGet
*GenesApi* | [**genesEntrezGeneIdVariantsGetUsingGET**](docs/GenesApi.md#genesEntrezGeneIdVariantsGetUsingGET) | **GET** /genes/{entrezGeneId}/variants | genesEntrezGeneIdVariantsGet
*GenesApi* | [**genesGetUsingGET**](docs/GenesApi.md#genesGetUsingGET) | **GET** /genes | genesGet
*GenesApi* | [**genesLookupGetUsingGET**](docs/GenesApi.md#genesLookupGetUsingGET) | **GET** /genes/lookup | genesLookupGet
*GenesetsApi* | [**genesetsGetUsingGET**](docs/GenesetsApi.md#genesetsGetUsingGET) | **GET** /genesets | genesetsGet
*GenesetsApi* | [**genesetsUuidGetUsingGET**](docs/GenesetsApi.md#genesetsUuidGetUsingGET) | **GET** /genesets/{uuid} | genesetsUuidGet
*InfoApi* | [**infoGetUsingGET**](docs/InfoApi.md#infoGetUsingGET) | **GET** /info | infoGet
*LevelsApi* | [**levelsGetUsingGET**](docs/LevelsApi.md#levelsGetUsingGET) | **GET** /levels | levelsGet
*LevelsApi* | [**levelsResistanceGetUsingGET**](docs/LevelsApi.md#levelsResistanceGetUsingGET) | **GET** /levels/resistance | levelsResistanceGet
*LevelsApi* | [**levelsSensitiveGetUsingGET**](docs/LevelsApi.md#levelsSensitiveGetUsingGET) | **GET** /levels/sensitive | levelsSensitiveGet
*SearchApi* | [**searchGetUsingGET**](docs/SearchApi.md#searchGetUsingGET) | **GET** /search | searchGet
*SearchApi* | [**searchPostUsingPOST**](docs/SearchApi.md#searchPostUsingPOST) | **POST** /search | searchPost
*TrialsApi* | [**trialsGetByCancerTypesUsingPOST**](docs/TrialsApi.md#trialsGetByCancerTypesUsingPOST) | **POST** /trials/cancerTypes | Return a list of trials using cancer types
*TrialsApi* | [**trialsMatchingGetUsingGET**](docs/TrialsApi.md#trialsMatchingGetUsingGET) | **GET** /trials | Return a list of trials using OncoTree Code and/or treatment
*VariantsApi* | [**utilsAllActionableVariantsGetUsingGET**](docs/VariantsApi.md#utilsAllActionableVariantsGetUsingGET) | **GET** /utils/allActionableVariants | utilsAllActionableVariantsGet
*VariantsApi* | [**utilsAllActionableVariantsTxtGetUsingGET**](docs/VariantsApi.md#utilsAllActionableVariantsTxtGetUsingGET) | **GET** /utils/allActionableVariants.txt | utilsAllActionableVariantsTxtGet
*VariantsApi* | [**utilsAllAnnotatedVariantsGetUsingGET**](docs/VariantsApi.md#utilsAllAnnotatedVariantsGetUsingGET) | **GET** /utils/allAnnotatedVariants | utilsAllAnnotatedVariantsGet
*VariantsApi* | [**utilsAllAnnotatedVariantsTxtGetUsingGET**](docs/VariantsApi.md#utilsAllAnnotatedVariantsTxtGetUsingGET) | **GET** /utils/allAnnotatedVariants.txt | utilsAllAnnotatedVariantsTxtGet
*VariantsApi* | [**variantsGetUsingGET**](docs/VariantsApi.md#variantsGetUsingGET) | **GET** /variants | variantsGet
*VariantsApi* | [**variantsLookupGetUsingGET**](docs/VariantsApi.md#variantsLookupGetUsingGET) | **GET** /variants/lookup | variantsLookupGet
*VariantsApi* | [**variantsLookupPostUsingPOST**](docs/VariantsApi.md#variantsLookupPostUsingPOST) | **POST** /variants/lookup | variantsLookupPost


## Documentation for Models

 - [ActionableGene](docs/ActionableGene.md)
 - [Alteration](docs/Alteration.md)
 - [AnnotateCopyNumberAlterationQuery](docs/AnnotateCopyNumberAlterationQuery.md)
 - [AnnotateMutationByGenomicChangeQuery](docs/AnnotateMutationByGenomicChangeQuery.md)
 - [AnnotateMutationByHGVSgQuery](docs/AnnotateMutationByHGVSgQuery.md)
 - [AnnotateMutationByProteinChangeQuery](docs/AnnotateMutationByProteinChangeQuery.md)
 - [AnnotateStructuralVariantQuery](docs/AnnotateStructuralVariantQuery.md)
 - [AnnotatedVariant](docs/AnnotatedVariant.md)
 - [Arms](docs/Arms.md)
 - [Article](docs/Article.md)
 - [ArticleAbstract](docs/ArticleAbstract.md)
 - [CancerGene](docs/CancerGene.md)
 - [CancerTypesQuery](docs/CancerTypesQuery.md)
 - [Citations](docs/Citations.md)
 - [CuratedGene](docs/CuratedGene.md)
 - [Drug](docs/Drug.md)
 - [Evidence](docs/Evidence.md)
 - [EvidenceQueries](docs/EvidenceQueries.md)
 - [EvidenceQueryRes](docs/EvidenceQueryRes.md)
 - [Gene](docs/Gene.md)
 - [GeneEvidence](docs/GeneEvidence.md)
 - [Geneset](docs/Geneset.md)
 - [Implication](docs/Implication.md)
 - [IndicatorQueryResp](docs/IndicatorQueryResp.md)
 - [IndicatorQueryTreatment](docs/IndicatorQueryTreatment.md)
 - [InfoLevel](docs/InfoLevel.md)
 - [MainType](docs/MainType.md)
 - [MutationEffectResp](docs/MutationEffectResp.md)
 - [OncoKBInfo](docs/OncoKBInfo.md)
 - [Query](docs/Query.md)
 - [QueryGene](docs/QueryGene.md)
 - [ResponseEntity](docs/ResponseEntity.md)
 - [Treatment](docs/Treatment.md)
 - [TreatmentDrug](docs/TreatmentDrug.md)
 - [TreatmentDrugId](docs/TreatmentDrugId.md)
 - [Trial](docs/Trial.md)
 - [TumorType](docs/TumorType.md)
 - [VariantConsequence](docs/VariantConsequence.md)
 - [VariantSearchQuery](docs/VariantSearchQuery.md)
 - [Version](docs/Version.md)


## Documentation for Authorization

Authentication schemes defined for the API:
### Bearer

- **Type**: API key
- **API key parameter name**: Bearer
- **Location**: HTTP header


## Recommendation

It's recommended to create an instance of `ApiClient` per thread in a multithreaded environment to avoid any potential issues.

## Author

contact@oncokb.org

